packages/paima-sdk/paima-events/src/builtin-events.ts

```diff
@@ -68,9 +68,20 @@ const RollupBlock = genEvent({
   ],
 });
 
+const BatcherPayment = genEvent({
+  name: 'BatcherPayment',
+  fields: [
+    { name: 'batcherAddress', type: Type.String(), indexed: true },
+    { name: 'userAddress', type: Type.String(), indexed: true },
+    { name: 'operation', type: Type.Union([Type.Literal('payGas'), Type.Literal('addFunds')]) },
+    { name: 'wei', type: Type.String() },
```

Comment:
`wei` is EVM-specific, but the batcher is not just for EVM

There is maybe a way to make this more generic by adding the caip2 prefix, but you could also have different events for different stacks as well

Reply:
I updated the name to a more generic 'value'. As the batcher wallet has to be in the same network as the PaimaL2Contract, I think the caip2 might be redundant?